Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm

```diff
@@ -187,8 +196,12 @@
 
     m_webPageProxy.layerTreeCommitComplete();
 
-    if (std::exchange(m_didUpdateMessageState, NeedsDidUpdate) == MissedCommit)
-        didRefreshDisplay();
+    if (!layerTreeTransaction.isMainFrameProcessTransaction() || std::exchange(m_didUpdateMessageState, NeedsDidUpdate) == MissedCommit) {
```

Comment:
This turned something quite simple into something more complicated that hard for me for wrap my head around.

First, let's factor non-main-frame commit stuff into new functions on this class.

Second, I think the completion handler is called synchronously? It would be easier to understand without using a completion handler.